front/components/assistant_builder/actions/MCPAction.tsx

```diff
@@ -99,8 +100,8 @@ export function MCPAction({
     );
 
   // MCPServerView on default MCP server will not allow switching to another one.
-  const isDefaultMCPServer = useMemo(
-    () => !!selectedMCPServerView?.server.isDefault,
+  const selectedServerAvailability: MCPServerAvailability | null = useMemo(
```

Comment:
Please check the changes on this file carefully because I did not understand exactly the conditions below that was partly using `selectedMCPServerView` and partly the memo'ed `isDefaultMCPServer`. 
I tried to clarify but I want to make sure I got it right 🙏🏻